lib/autoload/trackable_job.rb

```diff
@@ -22,13 +22,51 @@ class Job < ApplicationRecord
     validates :error, absence: true, unless: :errored?
     validates :status, presence: true
 
+    def in_queue?
+      # Only check sidekiq when some time has passed since the job was created.
+      return true if created_at > Time.zone.now - 5.minutes
+
+      job_in_sidekiq?
+    end
+
     private
 
     def signal_finished
       return unless saved_change_to_status?
 
       execute_after_commit { signal }
     end
+
+    def job_in_sidekiq?
+      return true unless Rails.env.production?
+
+      check_sidekiq_workers || check_sidekiq_queues
+    end
+
+    def check_sidekiq_workers
+      workers = Sidekiq::Workers.new
+      job_ids = workers.map do |_, _, work|
```

Comment:
Would this be equivalent to using `any?`?

```rb
workers.any? do |_, _, work|
  payload = JSON.parse(work['payload'])
  payload['args']&.first['job_id'] == id
end
```

If yes, this should be faster with one less iteration and the possibility of an early return.

Ditto for `check_sidekiq_queues` below.